### PR TITLE
Enable nullability in MouseHoverTimer

### PR DESCRIPTION
**Pull request by gpetrou**
## Proposed changes

- Enable nullability in MouseHoverTimer.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/6395)

### CODE REVIEW

File: src/System.Windows.Forms/src/System/Windows/Forms/MouseHoverTimer.cs [hunk 1]
```diff
@@ -2,16 +2,14 @@
...
 // See the LICENSE file in the project root for more information.
 
-#nullable disable
-
 namespace System.Windows.Forms
 {
     internal class MouseHoverTimer : IDisposable
     {
-        private Timer _mouseHoverTimer = new Timer();
+        private Timer? _mouseHoverTimer = new Timer();
```
Comment (dreddy-work) [inline src/System.Windows.Forms/src/System/Windows/Forms/MouseHoverTimer.cs:9]: why?

File: src/System.Windows.Forms/src/System/Windows/Forms/MouseHoverTimer.cs [hunk 2]
```diff
@@ -27,22 +25,26 @@ public void Start(ToolStripItem item)
             }
 
             _currentItem = item;
```
Comment (dreddy-work) [inline src/System.Windows.Forms/src/System/Windows/Forms/MouseHoverTimer.cs:27]: `item `should be nullable here.

**Review by Tanya-Solyanik — APPROVED** [review]
Thank you!

File: src/System.Windows.Forms/src/System/Windows/Forms/MouseHoverTimer.cs [hunk 3]
```diff
@@ -2,47 +2,49 @@
...
 // See the LICENSE file in the project root for more information.
 
-#nullable disable
-
 namespace System.Windows.Forms
 {
     internal class MouseHoverTimer : IDisposable
     {
-        private Timer _mouseHoverTimer = new Timer();
+        private Timer? _mouseHoverTimer = new Timer();
```
Comment (dreddy-work) [inline src/System.Windows.Forms/src/System/Windows/Forms/MouseHoverTimer.cs:9]: @gpetrou , the way this is initialized, we do not expect `_mouseHoverTimer` to be null in any practical scenarios here. I understand that `null` assignment statement in the `dispose ` method is forcing this.  We think the right thing to do here is remove the `null `assignment statement in the dispose method and keep this non-nullable.

**Review by dreddy-work — CHANGES_REQUESTED** [review]

**Review by dreddy-work — APPROVED** [review]

**Review by Tanya-Solyanik — APPROVED** [review]
Thank you!

**Pull request merged**